### ISSUE TITLE
Error during chatwoot_prepare

### ISSUE THREAD

**Issue by tiagominatel** [chatwoot/chatwoot#3638]
Hello guys,

i'm following help center's instructions to set up a self hosted chatwoot using docker, in my company we have separated cloud structures for database so i'm not using postgres or redis as containers, plus i can't have a superadmin user on postgres. 
All that said i got the follow error while running `docker-compose run --rm rails bundle exec rails db:chatwoot_prepare` :

```
...
+ exec bundle exec rails db:chatwoot_prepare
Rspec gem is missing. Install it.
Flay gem is missing. Install it.
Rspec gem is missing. Install it.
I, [2021-12-21T14:53:30.685660 #1]  INFO -- : Migrating to CreateInstallationConfig (20200510112339)
== 20200510112339 CreateInstallationConfig: migrating =========================
-- create_table(:installation_configs)
   -> 0.0423s
-- add_index(:installation_configs, [:name, :created_at], {:unique=>true})
   -> 0.0089s
rails aborted!
StandardError: An error has occurred, this and all later migrations canceled:

undefined local variable or method `locked' for #<InstallationConfig id: nil, name: "INSTALLATION_NAME", serialized_value: "{}", created_at: nil, updated_at: nil>
Did you mean?  lock!
/app/app/models/installation_config.rb:41:in `set_lock'
/app/lib/config_loader.rb:62:in `save_as_new_config'
/app/lib/config_loader.rb:52:in `save_general_config'
/app/lib/config_loader.rb:43:in `block in reconcile_general_config'
/app/lib/config_loader.rb:40:in `each'
/app/lib/config_loader.rb:40:in `reconcile_general_config'
/app/lib/config_loader.rb:23:in `process'
/app/db/migrate/20200510112339_create_installation_config.rb:11:in `change'
/app/lib/tasks/db_enhancements.rake:27:in `block (3 levels) in <main>'
/app/lib/tasks/db_enhancements.rake:18:in `each'
/app/lib/tasks/db_enhancements.rake:18:in `block (2 levels) in <main>'

Caused by:
NameError: undefined local variable or method `locked' for #<InstallationConfig id: nil, name: "INSTALLATION_NAME", serialized_value: "{}", created_at: nil, updated_at: nil>
Did you mean?  lock!
/app/app/models/installation_config.rb:41:in `set_lock'
/app/lib/config_loader.rb:62:in `save_as_new_config'
/app/lib/config_loader.rb:52:in `save_general_config'
/app/lib/config_loader.rb:43:in `block in reconcile_general_config'
/app/lib/config_loader.rb:40:in `each'
/app/lib/config_loader.rb:40:in `reconcile_general_config'
/app/lib/config_loader.rb:23:in `process'
/app/db/migrate/20200510112339_create_installation_config.rb:11:in `change'
/app/lib/tasks/db_enhancements.rake:27:in `block (3 levels) in <main>'
/app/lib/tasks/db_enhancements.rake:18:in `each'
/app/lib/tasks/db_enhancements.rake:18:in `block (2 levels) in <main>'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)
```
Is that error related to the user not having enough privileges on postgres? In that case, is there any workarround, maybe some script that i can suplly to DBA to prepare the database in advance?

**Comment by rahulmr** [comment]
Got same error in the develop branch, when I checked out master branch it worked fine.

**Comment by sojan-official** [comment]
This error occurs when the chatwoot rails application attempts to run rails migrations right from the start. Some of the older migrations would then fail because the underlying data models changed as the code base evolved. 

So when setting up an existing rails application into a new environment, it is recommended to do a `schema:load`  instead of running full migrations. 

ref: https://axiomq.com/blog/rake-dbschemaload-vs-rake-dbmigrate/

----


`db:chatwoot_prepare` aims to solve this choice for the end-users of chatwoot. it checks whether `ar_internal_metadata` table exists ( means that the database set up was already done )

if it doesn't exist -> do set up using schema:load 
if it exists  -> fall back to the remaining migrations and do just db:migrate

https://github.com/chatwoot/chatwoot/blob/28da30f7e40be02476b37881c3884ad0cacd1fbe/lib/tasks/db_enhancements.rake#L21


In your case it must be that the `ar_internal_metadata` table exists for some reason but the `schema:load` wasn't executed. 

----

## TLDR: 

run 
```
rake db:schema:load
rake db:seed
```

@tiagominatel  @rahulmr

**Comment by rahulmr** [comment]
Thanks @sojan-official will try it out.

**Comment by tiagominatel** [comment]
Thanks @sojan-official it worked! 🎉

**Comment by sojan-official** [comment]
@tiagominatel  @rahulmr  ref:

**Comment by YeonV** [comment]
Same error occurs to me....
How/where do i run those commands exactly?

![image](https://user-images.githubusercontent.com/28861537/147765687-025c7837-a427-4eeb-a144-a39bc36eb509.png)


This issue should be referenced here:
https://www.chatwoot.com/docs/self-hosted/deployment/docker

**Comment by YeonV** [comment]
> Thanks @sojan-official it worked! 🎉

How bro? could you please share the exact steps? <3
